### ISSUE TITLE
`tensor.exponential_()` samples 0

### ISSUE THREAD

**Issue by taylanbil** [pytorch/xla#2562]
## 🐛 Bug

`tensor.exponential_()` occasionally returns 0 and that causes problems downstream.

## To Reproduce

```python

def main_exp(dev, seed=123):

    def markstep(dev):
        if dev == 'tpu':
            xm.mark_step()

    torch.manual_seed(seed)
    if dev == 'tpu':
        dev = xm.xla_device()
    t1 = torch.randn([2048,2048], device=dev)
    markstep(dev)
    N = 100
    for i in range(0, N):
        m = -torch.empty_like(
                t1, memory_format=torch.legacy_contiguous_format,
                device=dev,
            ).exponential_().min().cpu().item()
        markstep(dev)
        if m == 0:
            print('on device {}, min was 0 after {} tries!!!'.format(dev, i))
            break
    else:
        print('on device {}, min was never 0 after {} tries'.format(dev, N))
```

```bash
$ python ./gumbel.py tpu
on device xla:1, min was 0 after 1 tries!!!
$ python ./gumbel.py cpu
on device cpu, min was never 0 after 100 tries
```

## Expected behavior

Exponential distribution should never return 0.

## Environment

 - Reproducible on XLA backend [CPU/TPU]: TPU
 - torch_xla version: 1.6, probably on nightly too.

## Additional context

This blocks wav2vec2.0, which uses `F.gumbel_softmax` [here](https://github.com/pytorch/fairseq/blob/master/fairseq/modules/gumbel_vector_quantizer.py#L170). gumbel softmax uses exponential().log(), which creates infinity, which makes softmax nan. In turn, this makes the loss nan.

**Comment by JackCaoG** [comment]
I was able to repo this problem, the problem comes from our lowering

https://github.com/pytorch/xla/blob/df0a2776e42ab144cfb00485c50852ea70f2b9b8/torch_xla/csrc/xla_lower_util.cpp#L483

We used `uniform distribution(0, 1-eps)` and `-log(1 - uniform) / lambda ` to generate the exponential (reference in [here](https://stats.stackexchange.com/questions/234544/from-uniform-distribution-to-exponential-distribution-and-vice-versa)). When the `uniform == 1`, `-log(1 - uniform) / lambda  == 0` . We should change the range to `uniform distribution(eps, 1-eps)` to avoid this.
